cli/src/commands/extensions/api.rs

```diff
@@ -47,35 +48,41 @@ impl<T: Unpin> OnceFuture<T> {
 }
 
 /// Opaquely encapsulates the extension state.
-pub struct ExtensionState(OnceFuture<Result<Rc<PhylumApi>>>);
+pub struct ExtensionState(Mutex<OnceFuture<Result<Rc<PhylumApi>>>>);
 
 impl From<BoxFuture<'static, Result<PhylumApi>>> for ExtensionState {
     fn from(extension_state_future: BoxFuture<'static, Result<PhylumApi>>) -> Self {
-        Self(OnceFuture::new(Box::pin(async {
+        Self(Mutex::new(OnceFuture::new(Box::pin(async {
             extension_state_future.await.map(Rc::new)
-        })))
+        }))))
+    }
+}
+
+impl ExtensionState {
+    async fn get(&self) -> Result<Rc<PhylumApi>> {
+        // The mutex guard is only useful for synchronizing access to the encapsulated future.
+        // Once a `Result<Rc<PhylumApi>>` is obtained, the guard is dropped: subsequent awaits on
+        // `PhylumApi` methods are not synchronized via this mutex, and can happen concurrently.
```

Comment:
Saying this mutex is only useful for synchronizing access is kinda misleading. I was going to suggest you'd use a pure synchronization primitive instead until I realized it's not just for synchronization, it's also necessary because we need interior mutability.

Reply:
Wording these comments is kind of tricky. Interior mutability is needed because we need to synchronize the mutable access to `OnceFuture`. I'll try and reword this.